lib/_http_client.js

```diff
@@ -639,7 +639,9 @@ ClientRequest.prototype.setTimeout = function(msecs, callback) {
   }
 
   this.once('socket', function(sock) {
-    sock.setTimeout(msecs, emitTimeout);
+    sock.once('connect', function() {
+      this.setTimeout(msecs, emitTimeout);
```

Comment:
:smile: … yeah, if you want to leave it at `sock.setTimeout` for readability, I’m fine with that, too. ;)